terraform/modules/s3/scripts/s3management.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Variables
+MODEL_NAME=""
+MINIO_APP=""
+MC_BINARY_URL=""
+MINIO_URL=""
+MINIO_USER="${MINIO_USER:-}"
+MINIO_PASSWORD="${MINIO_PASSWORD:-}"
+LOKI_BUCKET=""
+MIMIR_BUCKET=""
+TEMPO_BUCKET=""
+LOKI_INTEGRATOR=""
+MIMIR_INTEGRATOR=""
+TEMPO_INTEGRATOR=""
+
+MC_BINARY="/root/minio/mc"
+
+# Parse arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --model-name) MODEL_NAME="$2"; shift ;;
+    --minio-app) MINIO_APP="$2"; shift ;;
+    --mc-binary-url) MC_BINARY_URL="$2"; shift ;;
+    --minio-url) MINIO_URL="$2"; shift ;;
+    --minio-user) MINIO_USER="$2"; shift ;;
+    --minio-password) MINIO_PASSWORD="$2"; shift ;;
+    --loki-bucket) LOKI_BUCKET="$2"; shift ;;
+    --mimir-bucket) MIMIR_BUCKET="$2"; shift ;;
+    --tempo-bucket) TEMPO_BUCKET="$2"; shift ;;
+    --loki-integrator) LOKI_INTEGRATOR="$2"; shift ;;
+    --mimir-integrator) MIMIR_INTEGRATOR="$2"; shift ;;
+    --tempo-integrator) TEMPO_INTEGRATOR="$2"; shift ;;
+    *) echo "Unknown parameter passed: $1"; exit 1 ;;
+  esac
+  shift
+done
+
+# Credentials validation
+if [[ -z "$MINIO_USER" || -z "$MINIO_PASSWORD" ]]; then
+  echo "Error: MINIO_USER and MINIO_PASSWORD must be set either as arguments or environment variables."
+  exit 1
+fi
+
+
+# Functions
+wait_for_app() {
+  local app="$1"
+  echo "Waiting for application $app in model $MODEL_NAME..."
+  juju wait-for application "$app" -m "$MODEL_NAME" --timeout 20m
+}
+
+bucket_exists() {
+  local bucket_name="$1"
+  juju ssh -m "$MODEL_NAME" "$MINIO_APP/leader" "$MC_BINARY" ls local/ | grep -q "$bucket_name"
+}
```

Comment:
Are we supposed to `echo $?` at the end?

Reply:
It is not necessary:

```shell
╭─ubuntu@charm-dev-35 ~ [microk8s:test]
╰─$ juju ssh minio/leader /root/minio/mc ls local/ | grep -q "loki" 
╭─ubuntu@charm-dev-35 ~ [microk8s:test]
╰─$ echo $?                                                        
0

╭─ubuntu@charm-dev-35 ~ [microk8s:test]
╰─$ juju ssh minio/leader /root/minio/mc ls local/ | grep -q "Leon"
╭─ubuntu@charm-dev-35 ~ [microk8s:test]
╰─$ echo $?                                                                                                                               
1
```